### ISSUE TITLE
Suggestion / Feature request Automatic Price corrections (something like a Playbook)

### ISSUE THREAD

**Issue by xaraxarax** [rgriebl/brickstore#588]
(Not) Every time I update the prices, I think, couldn't be easier to add some spice to the price. If a table (perhaps in the settings) contains some information like category or item and some adjustments like percentage price corrections or fixed surcharges or discounts, this could easily be implemented. And sometimes I think this can be extended to set some additional values ​​like comments and remarks.

What do you think about it, could something like this be implemented and would others also find it nice?

**Comment by paramecie** [comment]
Based on "some information" and "some adjustement"...
If think this is a very too generic demand :-(

And any fixed solution would be fine for some people, while another would be prefered by the others...
I can't see any consensus on this.

You can already do what you wish manually.
Like using the Filter and limit on "Brick" Category, CTRL+A, then decrease their price of 10% for example.
Or filtering on price > 1$ and adding 0.10$ on all of them.
Or sorting by Color and adding 5% to all Black lots and removing 10% to all Yellow...

In short you're free to do what you wish without anything imposed.
Hoping it helps?

**Comment by xaraxarax** [comment]
With this argument you question the existence of Brickstore itself, because every goal that you can achieve with Brickstore, you will certainly also achieve directly with Bricklink, you just have to click and filter a few times and so on. But we want all go the easy way and therefore we use Brickstore. I don't see it being a fixed solution that requires consensus. It can be a table that everyone fills with their own wishes or leaves empty. And then when you part out you just choose whether you want to apply these adjustments or not.

But I also have a real world example for you. We part out sets daily and the workflow is mostly the same - similar to what you described.

- we sort parts
- delete things that we cannot sell individually (stickers, large parts such as aircraft chassis, vehicle parts, etc.)
- download the current prices (average sale 6m) and set my purchasing costs
- increase prices for parts by 10%, for minifigures by 20%
- Adjust surcharges and discounts for certain categories of parts
- adjust surcharges and reductions individually for certain parts
- set annotations for some categories (MF and Tiles) and individually for some parts with shiny surfaces

What would also add value for me would be to adjust the prices only if the price is not lower than a certain price in relation to purchasing costs. 

Unfortunately, mistakes happen from time to time, and steps are forgotten or values ​​​​set incorrectly. Doesn't everyone have an equally good day. For me, one of the biggest advantages of automation is error reduction, time saving and increase the economically viable. Therefore my wishful thinking to automate all these steps. But of course it is clear to me that a) there would have to be more interested parties and b) the will on the part of the developer must also be there. That's why it's just a suggestion and wishful thinking :)

**Comment by paramecie** [comment]
>With this argument you question the existence of Brickstore itself, because every goal that you can achieve with Brickstore, you will certainly also achieve directly with Bricklink, you just have to click and filter a few times and so on.

Sorry it's wrong. There are tens of features BS does that BL don't - at all - whatever the number of clicks.

>if the price is not lower than a certain price in relation to purchasing costs.

Or for other users;
* if the qty is above a number
* if the qty is below a number
* if the price is...
* if the last sold date is before (or after)...
* if the weight of the part is below (or above)...
* if the category is/isn't
* if the type is/isn't
* if the color is/isn't
* if there is/not a comment or a remark
* if it has/not bulk quantity
* if it has a lot ID
* if....

Plus all those combinations with OR and AND.

Like: if not a Minifigure AND (more than 100 qty OR less than 0.02$ price) AND color is not Red AND doesn't have a Comment AND have a lot ID AND not sold since 01/01/2022

Ouch

What I mean, is that it'd require a fully programmable feature or a grid of parameters a big as your screen :-)
Which will always the one this user would need...

It's interesting but it's a full new and big feature - or some way to program BrickStore.

**Comment by ZZJHONS** [comment]
Maybe you can use the Extensions interface: https://rgriebl.github.io/brickstore/extensions/

**Comment by paramecie** [comment]
Yes, I was wondering if it could be possible...
Never checked, I've to read a bit about this.

**Comment by xaraxarax** [comment]
> Sorry it's wrong. There are tens of features BS does that BL don't - at all - whatever the number of clicks.

I never doubted, but the goal is always the same, to list parts and prices on Bricklink. And I can do that either old school with bricklink, faster with Brickstore and even faster with Brickstore with automatisms.

> [...] Ouch

I see your arguments, but firstly you can start small and achieve big effects and secondly there is already a strong filter engine (the text filter is my favorite) that can perhaps also be used here, when it has to become more complex at some point.

> Maybe you can use the Extensions interface: https://rgriebl.github.io/brickstore/extensions/

I don't know how many times I've landed on the site, but every time I look for an example or a clue as to where the learning curve needs to start and ... fail

I don't get into the details of the code, i don't speak the language(s) and have no idea how i can teach myself both in a timely manner.
